src/utils/auth.ts

```diff
@@ -11,10 +11,13 @@ const authenticateJWT = async (req: Request, res: Response, next: NextFunction)
     if (authHeader && authHeader.startsWith("Bearer ")) {
         const token = authHeader.split(' ')[1];
         try {
-            const {id} = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
+            const user = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
+            
 
-            if (id) {
-                req.user = {id: Number(id)};
+            if (user) {
+                // TODO: This assignment was changed from {id: Number(id)} for user bc TS wanted all 
+                // properties defined on User type. Is this OK?
+                req.user = user;
```

Comment:
I suspect this isn't doing what you think it is - I believe that jwt.verify is returning type `any` which effectively tells TS to not pay attention to type of this variable at all.

If you want to return the full user object (what I believe you intended?), I recommend instead actually fetching the user by ID using prisma

Reply:
Made an Issue to address this